lib/capistrano/tasks/docker.rake

```diff
@@ -35,21 +45,25 @@ namespace :docker do
   task :build => [:'docker:update_mirror'] do
     on fetch(:docker_build_server_host) do
       within dockerbuild_plugin.docker_build_base_path do
-        timestamp = Time.now.to_i
-        git_sha1 = `git rev-parse #{fetch(:branch)}`.chomp
-        worktree_dir_name = "worktree-#{git_sha1}-#{timestamp}"
+        if fetch(:docker_build_no_worktree)
+          execute(:git, :reset, "--hard", fetch(:branch))
```

Comment:
I think we should introduce a lock file as follows:

```ruby
# `lock_fiile` is something like "#{File.join(Dir.tmpdir, CGI.escape(repo_url))}.lock"
lock_file = dockerbuild_plugin.lock_file_path(repo_url)
if File.exist?(lock_file) && File.read(lock_file) != fetch(:branch)
  raise "This repository is locked."
end

upload!(StringIO.new(fetch(:branch)), lock_file) # This file should be deleted in the ensure block
```

Reply:
I use `flock` command that waits for the release of the lock file automatically.